### PR TITLE
Changes for MROT

### PR DESCRIPTION
**Pull request by jayeshh123**
This PR is related to MROT configuration.

Hi @sasikeda Request you to review this PR. Thank you.

### CODE REVIEW

File: ibmcloud_scale_templates/sub_modules/instance_template/variables.tf [hunk 1]
```diff
@@ -296,3 +296,15 @@ variable "bastion_user" {
   default     = "ubuntu"
   description = "Provide the username for Bastion login."
 }
+
```
Comment (digvijay-ukirde) [inline ibmcloud_scale_templates/sub_modules/instance_template/variables.tf]: Can we get the CIDR by vpc_storage_cluster_private_subnets?
https://github.com/IBM/ibm-spectrum-scale-cloud-install/blob/next_gen/ibmcloud_scale_templates/sub_modules/instance_template/variables.tf#L32

File: ibmcloud_scale_templates/sub_modules/instance_template/variables.tf [hunk 2]
```diff
@@ -296,3 +296,15 @@ variable "bastion_user" {
...
+variable "storage_subnet_cidr" {
+  type        = string
+  default     = ""
+  description = "subnet cidr for storage cluster"
+}
+
+variable "compute_subnet_cidr" {
```
Comment (digvijay-ukirde) [inline ibmcloud_scale_templates/sub_modules/instance_template/variables.tf]: Same as storage.

File: resources/ibmcloud/compute/vsi_0_vol/vsi_0_vol.tf [hunk 3]
```diff
@@ -40,9 +46,11 @@ then
     then
         PACKAGE_MGR=dnf
         package_list="python38 kernel-devel-$(uname -r) kernel-headers-$(uname -r)"
+        sudo dnf install firewalld -y
```
Comment (digvijay-ukirde) [inline resources/ibmcloud/compute/vsi_0_vol/vsi_0_vol.tf]: No sudo is needed and you can add the firewall package as a part of the package_list.

File: resources/ibmcloud/compute/vsi_0_vol/vsi_0_vol.tf [hunk 4]
```diff
@@ -106,6 +114,36 @@ firewall-offline-cmd --zone=public --add-port=9085/tcp
...
+        sed -i '/DOMAIN/d' /etc/sysconfig/network-scripts/ifcfg-eth1
+        hwaddr=$(ifconfig | grep ether | awk -F " " '{print$2}' | awk 'NR==2 {print}')
+        sec_interface=$(nmcli -t con show --active | grep eth1 | cut -d ':' -f 1)
+        echo "HWADDR=$hwaddr" >> /etc/sysconfig/network-scripts/ifcfg-eth1
+        echo "NAME=\"$sec_interface\"" >> /etc/sysconfig/network-scripts/ifcfg-eth1
+        echo "DOMAIN=${var.storage_domain_name}" >> /etc/sysconfig/network-scripts/ifcfg-eth1
+        systemctl restart NetworkManager
+        sudo nmcli connection up "$sec_interface"
+    else
+        cp /etc/sysconfig/network-scripts/ifcfg-eth0 /etc/sysconfig/network-scripts/ifcfg-eth1
```
Comment (digvijay-ukirde) [inline resources/ibmcloud/compute/vsi_0_vol/vsi_0_vol.tf]: What's the diff between if and else stanzas?

Comment (digvijay-ukirde) [reply resources/ibmcloud/compute/vsi_0_vol/vsi_0_vol.tf]: Also, can you try using nmcli to generate the file?
`nmcli conn del eth1`
`nmcli con add type ethernet con-name eth1 ifname eth1`
You should be able to add the domain and restart the service.

File: resources/common/scripts/prepare_remote_mount_inv.py [hunk 5]
```diff
@@ -118,6 +118,7 @@ def initialize_node_details(storage_gui_ip, user, key_file):
         "user": user,
         "key_file": key_file,
         "class": "storagenodegrp",
+        "host_name_eth1": storage_gui_ip.rsplit('.', 2)[0]
```
Comment (digvijay-ukirde) [inline resources/common/scripts/prepare_remote_mount_inv.py]: can we refactor the name to secondary_hostname (similar to secondary_interface) or something?

File: ibmcloud_scale_templates/sub_modules/instance_template/variables.tf (hunk 1 above)
Comment (jayeshh123) [reply ibmcloud_scale_templates/sub_modules/instance_template/variables.tf]: @digvijay-ukirde Resolved

File: ibmcloud_scale_templates/sub_modules/instance_template/variables.tf (hunk 2 above)
Comment (jayeshh123) [reply ibmcloud_scale_templates/sub_modules/instance_template/variables.tf]: Resolved

File: resources/ibmcloud/compute/vsi_0_vol/vsi_0_vol.tf (hunk 3 above)
Comment (jayeshh123) [reply resources/ibmcloud/compute/vsi_0_vol/vsi_0_vol.tf]: added package in package list

File: resources/ibmcloud/compute/vsi_0_vol/vsi_0_vol.tf (hunk 4 above)
Comment (jayeshh123) [reply resources/ibmcloud/compute/vsi_0_vol/vsi_0_vol.tf]: @digvijay-ukirde  Yes added nmcli command. As per this first deleting the network interface then adding it.

File: resources/common/scripts/prepare_remote_mount_inv.py (hunk 5 above)
Comment (jayeshh123) [reply resources/common/scripts/prepare_remote_mount_inv.py]: Name is changed

**Pull request merged**